pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -84,47 +87,62 @@ func getTLSConfigModification(getUpdateCreator secret.GetUpdateCreator, mdb mdbv
 		return automationconfig.NOOP(), nil
 	}
 
-	if err := ensureTLSSecret(getUpdateCreator, mdb); err != nil {
+	certificateKeyFileName, err := ensureTLSSecret(getUpdateCreator, mdb)
+	if err != nil {
 		return automationconfig.NOOP(), err
 	}
 
 	// The config is only updated after the certs and keys have been rolled out to all pods.
 	// The agent needs these to be in place before the config is updated.
 	// Once the config is updated, the agents will gradually enable TLS in accordance with: https://docs.mongodb.com/manual/tutorial/upgrade-cluster-to-ssl/
 	if hasRolledOutTLS(mdb) {
-		return tlsConfigModification(mdb), nil
+		return tlsConfigModification(mdb, certificateKeyFileName), nil
 	}
 
 	return automationconfig.NOOP(), nil
 }
 
 // ensureTLSSecret will create or update the operator-managed Secret containing
 // the concatenated certificate and key from the user-provided Secret.
-func ensureTLSSecret(getUpdateCreator secret.GetUpdateCreator, mdb mdbv1.MongoDB) error {
+func ensureTLSSecret(getUpdateCreator secret.GetUpdateCreator, mdb mdbv1.MongoDB) (string, error) {
```

Comment:
[nit] should we maybe rename the function to make clearer what it returns? It is not immediately obvious. However I am terrible at naming so I can't come up with anything better!

Reply:
Agreed but I'm horribly at naming as well. If the function is hard to name it might be because it does to much though. How about we split out the cert and key fetching to another function, like `func getCertAndKey(getter secret.Getter, mdb mdbv1.MongoDB) (string, string, error)` and then have the other function be: `func ensureTLSSecret(getUpdateCreator secret.GetUpdateCreator, mdbv1.MongoDB, cert, key string) error`?

Reply:
I think that would be a nice thing to do!

Reply:
Done!